### PR TITLE
refact(API): Adds missing input validations in all API methods and validates empty user Id.

### PR DESCRIPTION
**Pull request by rashidsp**
1. Aattributes validation in APIs(is_feature_enabled, get_enabled_features and get_feature_variables).
2. Allow empty user id.

### CODE REVIEW

**Comment by mikeng13** [comment]
build

**Comment by mikeng13** [comment]
build

File: lib/optimizely.rb [hunk 1]
```diff
@@ -96,11 +96,15 @@ def activate(experiment_key, user_id, attributes = nil)
 
       return nil unless Optimizely::Helpers::Validator.inputs_valid?(
         {
-          experiment_key: experiment_key,
-          user_id: user_id
+          experiment_key: experiment_key
         }, @logger, Logger::ERROR
       )
 
+      unless user_id.is_a?(String)
```
Comment (mikeng13) [inline lib/optimizely.rb]: I'd rather we refactor a bit and avoid using this same logic everywhere in the code. Let's mimic what the C# SDK does here: https://github.com/optimizely/csharp-sdk/blob/master/OptimizelySDK/Optimizely.cs#L642

Comment (rashidsp) [reply lib/optimizely.rb]: Done.

**Comment by msohailhussain** [comment]
it's ready for review.

File: lib/optimizely.rb [hunk 2]
```diff
@@ -258,6 +258,8 @@ def is_feature_enabled(feature_flag_key, user_id, attributes = nil)
         }, @logger, Logger::ERROR
       )
 
+      return false unless user_inputs_valid?(attributes)
```
Comment (mikeng13) [inline lib/optimizely.rb]: Can we combine `Optimizely::Helpers::Validator.inputs_valid?` with `user_inputs_valid?`? Seems like we could just put all user inputs into one dictionary and have them all validated

Comment (rashidsp) [reply lib/optimizely.rb]: Combined validations: https://github.com/optimizely/ruby-sdk/compare/rashid/input-validations...rashid/input-validations-2
It seems a review-able change plz confirm to continue with this PR?

**Pull request merged**